Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerTreeView.cpp

```diff
@@ -80,6 +80,11 @@ namespace AzToolsFramework
                 }
             }
         }
+
+        if (QModelIndex hoveredIndex = indexAt(m_mousePosition); m_currentHoveredIndex != hoveredIndex)
+        {
+            m_currentHoveredIndex = hoveredIndex;
```

Comment:
This still feels a bit brittle - do we need to cache the hovered index at all? It might be safer to recalculate. If we can't, I'd suggest using a `QPersistentModelIndex` that'll listen to all the model signals and not just `dataChanged`.

Reply:
I checked and it's actually only used in drawBranches, so there's not a huge advantage in caching the value in mouseMove. I'll just refactor it out and live with it :D

Reply:
Removed the cached value.